pkg/composer/composer.go

```diff
@@ -100,8 +100,11 @@ func applyAddPublicKeys(doc document.Document, publicKeys string) (document.Docu
 
 	existingPublicKeys := diddoc.PublicKeys()
 	for _, existing := range existingPublicKeys {
-		// if key already exists just replace it with old one
-		newPublicKeys[existing.ID()] = existing
+		// NOTE: If a key ID already exists, we will just replace the existing key
+		// so new public keys will retain new version
+		if _, ok := newPublicKeys[existing.ID()]; !ok {
```

Comment:
So you don't add it if it exists?

Reply:
No, you overwrite it so basically you don't touch the one new public keys (from the patch)

Reply:
you start with keys from the patch, and add keys from the document if that key is not in the patch (add list)